### DISCUSSION TITLE
How to access debugService from an Extension

### DISCUSSION THREAD

**Discussion by iAbadia** [microsoft/vscode-discussions#535] [Extension Development QnA] [answered]
Hi, I'm building an extension that needs to notify the Debugger to update its views (essentially re-evaluate watch expressions and variables). I see vscode itself accessing the debugService using an accessor but I can't seem to figure out a way for my extension to access services:

```
const debugService = accessor.get(IDebugService);
debugService.getViewModel().updateViews();
```

Is there a way to do this through the Extension API?

**Comment by isidorn** [comment, marked as answer]
This is currently not exposed through the Extension API. Some functionality is exposed through the `debug` namespace and I suggest you check that out.

**Comment by vmwsree** [comment]
I would also require this to be exposed for my extension, It will be helpful in the future if this is exposed
